### PR TITLE
[RFC] Deprecation and alternatives for utf8_encode and utf8_decode

### PR DESCRIPTION
**Pull request by IMSoP**
Proposed additional wording if https://wiki.php.net/rfc/remove_utf8_decode_and_utf8_encode passes

Builds on top of #1418

### CODE REVIEW

File: reference/iconv/functions/iconv.xml [hunk 1]
```diff
@@ -29,22 +28,22 @@
...
      <term><parameter>to_encoding</parameter></term>
      <listitem>
       <para>
-       The output charset.
+       The encoding which <parameter>string</parameter> should be converted to.
       </para>
       <para>
-       If you append the string <literal>//TRANSLIT</literal> to
+       If the string <literal>//TRANSLIT</literal> is appended to
        <parameter>to_encoding</parameter> transliteration is activated. This
```
Comment (Crell) [inline reference/iconv/functions/iconv.xml:43]:
```suggestion
       <parameter>to_encoding</parameter>, then transliteration is activated. This
```

File: reference/intl/uconverter/transcode.xml [hunk 2]
```diff
@@ -16,45 +16,58 @@
...
    </varlistentry>
    <varlistentry>
-    <term><parameter>toEncoding</parameter></term>
+    <term>
+     <parameter>toEncoding</parameter>
+    </term>
     <listitem>
      <para>
-      
+      The encoding which <parameter>str</parameter> should be converted to.
```
Comment (Crell) [inline reference/intl/uconverter/transcode.xml:42]:
```suggestion
      The encoding to which <parameter>str</parameter> should be converted.
```

File: reference/mbstring/functions/mb-convert-encoding.xml [hunk 3]
```diff
@@ -31,31 +30,36 @@
...
+       The &string; or &array; to be converted.
       </para>
      </listitem>
     </varlistentry>
     <varlistentry>
      <term><parameter>to_encoding</parameter></term>
      <listitem>
       <para>
-       The type of encoding that <parameter>string</parameter> is being converted to.
+       The encoding which <parameter>string</parameter> should be converted to.
```
Comment (Crell) [inline reference/mbstring/functions/mb-convert-encoding.xml:41]:
```suggestion
       The encoding to which <parameter>string</parameter> should be converted.
```

File: reference/strings/functions/utf8-decode.xml [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+     <![CDATA[
+sring(1) "?"
+]]>
+     </screen>
+    </informalexample>
+   </para>
+  </note>
+ </refsect1>
+
+ <refsect1 role="changelog">
```
Comment (Crell) [inline reference/strings/functions/utf8-decode.xml:207]: Should the deprecation be mentioned in the changelog, too?

File: reference/strings/functions/utf8-encode.xml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+  </methodsynopsis>
+  <para>
+   This function converts the string <parameter>string</parameter> from the
+   <literal>ISO-8859-1</literal> encoding to <literal>UTF-8</literal>.
+  </para>
+
+  <note>
+   <para>
+    This function does not attempt to guess the current encoding of the provided
+    string, it assumes it is encoded as ISO-8859-1 (also known as "Latin 1")
```
Comment (Crell) [inline reference/strings/functions/utf8-encode.xml:27]:
```suggestion
    string. It assumes it is encoded as ISO-8859-1 (also known as "Latin 1")
```

File: reference/intl/uconverter/transcode.xml (hunk 2 above)
Comment (IMSoP) [reply reference/intl/uconverter/transcode.xml:42]: Reading it back, all of these sentences were unnecessarily torturous. I've come up with a new wording across all three functions, which I think is less wordy and more precise.

File: reference/strings/functions/utf8-decode.xml (hunk 4 above)
Comment (IMSoP) [reply reference/strings/functions/utf8-decode.xml:207]: Yes, probably. Also, looks like I missed an attribute to make it list as deprecated in indexes.

File: reference/strings/functions/utf8-encode.xml (hunk 5 above)
Comment (IMSoP) [reply reference/strings/functions/utf8-encode.xml:27]: I compromised and used a semi-colon 😜

File: reference/intl/uconverter/transcode.xml (hunk 2 above)
Comment (IMSoP) [reply reference/intl/uconverter/transcode.xml:42]: Actually, that's part of https://github.com/php/doc-en/pull/1418 which has the changes which can land even if deprecation doesn't go ahead.

**Comment by IMSoP** [comment]
Sorry, I've made a mess of this; my intention was to compare two branches, but I ended up with two copies of each branch, confusing everything.

Re-opened here for now: https://github.com/IMSoP/doc-en/pull/1

**Pull request closed without merging**